native/cocos/renderer/gfx-agent/DeviceAgent.cpp

```diff
@@ -96,28 +96,33 @@ void DeviceAgent::doDestroy() {
         _mainMessageQueue, DeviceDestroy,
         actor, _actor,
         {
-            if (actor) {
-                actor->destroy();
-            }
+            actor->destroy();
         });
 
     if (_cmdBuff) {
-        static_cast<CommandBufferAgent *>(_cmdBuff.get())->destroyAgent();
+        static_cast<CommandBufferAgent *>(_cmdBuff)->destroyAgent();
+        static_cast<CommandBufferAgent *>(_cmdBuff)->_actor = nullptr;
+        delete _cmdBuff;
         _cmdBuff = nullptr;
     }
+    if (_queryPool) {
+        static_cast<QueryPoolAgent *>(_queryPool)->_actor = nullptr;
+        delete _queryPool;
+        _queryPool = nullptr;
+    }
+    if (_queue) {
+        static_cast<QueueAgent *>(_queue)->_actor = nullptr;
+        delete _queue;
+        _queue = nullptr;
+    }
 
-    _queryPool = nullptr;
-    _queue = nullptr;
+    _mainMessageQueue->terminateConsumerThread();
 
-    if (_mainMessageQueue) {
```

Comment:
Don't need to check whether _mainMessageQueue is valid now?

Reply:
I just revert the my modification. Previous implementation doesn't check it. I can add it back.

Reply:
I added it back.